cdc/owner/feed_state_manager.go

```diff
@@ -302,16 +337,40 @@ func (m *feedStateManager) handleError(errs ...*model.RunningError) {
 		}
 		for _, err := range errs {
 			info.Error = err
+			m.lastErrorTime = time.Now()
 			info.ErrorHis = append(info.ErrorHis, time.Now().UnixNano()/1e6)
 		}
-		changed := info.CleanUpOutdatedErrorHistory()
+		// if no errors occured in a time window, we can assume that changefeed
+		// was running normally before. So if we get an error for this changefeed now,
+		// we must reset the expBackoff and re-backoff from InitialInterval.
+		if !info.ErrorsReachedThreshold(m.backoffInterval) {
+			m.expBackoff.Reset()
+			m.backoffInterval = m.expBackoff.NextBackOff()
+		}
+		changed := info.CleanUpOutdatedErrorHistory(m.backoffInterval)
 		return info, changed || len(errs) > 0, nil
 	})
 
-	// if the number of errors has reached the error threshold, stop the changefeed
-	if m.state.Info.ErrorsReachedThreshold() {
-		m.shouldBeRunning = false
-		m.patchState(model.StateError)
-		return
+	if m.lastErrorTime != time.Unix(0, 0) {
+		if time.Since(m.lastErrorTime) < m.backoffInterval {
+			m.shouldBeRunning = false
+			m.patchState(model.StateError)
+		} else {
+			// clear lastErrorTime to give the changefeed another chance to run
+			m.lastErrorTime = time.Unix(0, 0)
+			m.backoffInterval = m.expBackoff.NextBackOff()
+
+			// if the duration since backoff start exceeds MaxElapsedTime,
+			// we set the state of changefeed to "failed" and don't let it run again unless it is manually resumed.
+			if m.backoffInterval == backoff.Stop {
+				log.Warn("changefeed will not be restarted because the duration since backoff start exceeds",
+					zap.Duration("max-elapsed-time", m.expBackoff.MaxElapsedTime))
+				m.shouldBeRunning = false
+				m.patchState(model.StateFailed)
+			} else {
+				log.Info("changefeed restart backoff interval is changed", zap.String("changefeedID", m.state.ID),
```

Comment:
Can this be a `Debug` log?

Reply:
> Can this be a `Debug` log?

This log is not printed frequently. So i think it's ok to use info-level info here.